profiles/common/modules/features/nexteuropa_newsroom/nexteuropa_newsroom.blocks.inc

```diff
@@ -0,0 +1,889 @@
+
+          case 'consultation_funding':
+            $default_view_display = 'consultation_funding';
+            break;
+        }
+      }
+
+      $newsroom_view_exec = _nexteuropa_newsroom_fetch_content_view($view_arguments, $newsroom_grouping_term, $default_view_display, $sticky, $full_display, $topic_operator);
+
+      // If we have some content, let's add a title.
+      if ($newsroom_view_exec && $newsroom_view_exec['block']['content'] != '') {
+
+        $topic_is_highlighted = _nexteuropa_newsroom_type_is_highlighted($newsroom_grouping_term);
+
+        if ($topic_is_highlighted) {
+          // Controls width.
+          $classes[] = 'featured';
+          // Controls border.
+          $classes[] = 'highlighted';
+        }
+
+        $clean_url_type_name = pathauto_cleanstring($newsroom_grouping_term->name);
+        $newsroom_block_id = 'newsroom-block-' . $clean_url_type_name;
+        if (count($topics) == 1) {
+          $topics_segment = pathauto_cleanstring($topics[key($topics)]->name);
+        }
+        else {
+          $topics_segment = $newsroom_topic_tids;
+        }
+
+        $full_list_url = NewsroomHelper::getNewsroomUrl($clean_url_type_name, $topics_segment);
+
+        // Check latest item in box.
+        $created = 0;
+
+        foreach ($newsroom_view_exec['view']->result as $row) {
+          $created = $row->node_created > $created ? $row->node_created : $created;
+        }
+
+        $newsroom_boxes[] = array_merge($newsroom_view_exec, array(
+          'title' => l($newsroom_grouping_term->name, $full_list_url),
+          'list_link' => l(t('<span class="more">More</span> <span class="more_type">%1</span>', array('%1' => $newsroom_grouping_term->name)), $full_list_url, array('html' => TRUE)),
+          'classes' => $classes,
+          'id' => $newsroom_block_id,
+          'highlight' => $topic_is_highlighted,
+          'agenda' => FALSE,
+          'last_update' => $created,
+          'weight' => $newsroom_grouping_term->weight,
+        ));
+      }
+    }
+  }
+
+  return !empty($newsroom_boxes) ? _nexteuropa_newsroom_render_boxes($newsroom_boxes) : '';
+}
+
+/**
+ * Render the boxes for the block.
+ *
+ * @return string
+ *   markup
+ */
+function _nexteuropa_newsroom_render_boxes($boxes) {
+  $newsroom_content = '';
+  $columns_newsroom = array(
+    1 => array(
+      'visible' => FALSE,
+      'content' => '',
+    ),
+    2 => array(
+      'visible' => FALSE,
+      'content' => '',
+    ),
+  );
+  $num_columns = variable_get('newsroom_summary_block_columns', 1);
+
+  // Sort boxes based on configuration.
+  uasort($boxes, '_nexteuropa_newsroom_sort_newsroom_boxes');
+
+  $newsroom_row_number = 1;
+  foreach ($boxes as $box) {
+    $active_column = 1;
+    $box['classes'][] = 'newsroom_item_container';
+    // Get active column.
+    if ($num_columns == 2 && $newsroom_row_number % 2 == 0) {
+      $active_column = 2;
+    }
+
+    $columns_newsroom[$active_column]['visible'] = TRUE;
```

Comment:
Please use a wrapper template instead of writing unmaintainable html source to a 'generator' function, <a> has to be implemented with l() and also the class shouldn't generated like it is here.